### PR TITLE
Add dedicated bencode.encodingLength() implementation

### PR DESCRIPTION
**Pull request by jhermsmeier**
This adds a standalone implementation for `.encodingLength(value)`,
which is significantly faster than encoding & getting the length from
the returned buffer.

This will also enable us to pre-allocate a correctly sized buffer when
encoding, potentially enabling performance improvements, as we can avoid
allocating a large amount of small buffers followed by concatenation,
and instead write directly into the pre-allocated buffer.

Further, `.encodingLength()` supports new types for lists and
dictionaries; namely `Set`s and `Map`s

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+var util = module.exports
+
+util.digitCount = function digitCount (value) {
+  // Add a digit for negative numbers, as the sign will be prefixed
+  var sign = value < 0 ? 1 : 0
+  // Guard against negative numbers & zero going into log10(),
+  // as that would return -Infinity
+  value = Math.abs(Number(value || 1))
```
Comment (themasch) [inline lib/util.js:8]: The ` ||  1`  here is just to guard againts null/undefined?

Comment (themasch) [reply lib/util.js:8]: nevermind, re-read the comment and got it! nice fix!

**Comment by jhermsmeier** [comment]
Looking at the CI builds, should we drop support for Node v6 and below? Considering that we're at v14 being LTS now, I think that should be safe :)

File: test/encoding-length.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+var fs = require('fs')
+var path = require('path')
+var test = require('tape').test
+var bencode = require('..')
+
+var torrent = fs.readFileSync(
+  path.join(__dirname, '..', 'benchmark', 'test.torrent')
+)
+
+test('encoding-length', function (t) {
```
Comment (themasch) [inline test/encoding-length.test.js:10]: could you maybe add a few simple test cases for non-empty lists/dicts? 
I know the `torrent` test case technically covers that, but I think it would be nice to see those in isolation to prevent a  case where multiple errors could cancel them out in a specific test file.

**Review by themasch — COMMENTED** [review]
would like to see two more test cases, other than that, excellent!

**Comment by themasch** [comment]
Welcome back! Nice to see someone still cares for this project :) 

> Looking at the CI builds, should we drop support for Node v6 and below? Considering that we're at v14 being LTS now, I think that should be safe :)

Yep, we should drop them... wow, time flies.

File: test/encoding-length.test.js (hunk 2 above)
Comment (jhermsmeier) [reply test/encoding-length.test.js:10]: Done!

**Comment by jhermsmeier** [comment]
> Welcome back! Nice to see someone still cares for this project :)

Thanks! Well, since the BitTorrent v2 spec dropped a little while ago, it had me wondering if this would still work in that context, and ended up spending yesterday going through all of it – there's more coming! :D

> Yep, we should drop them... wow, time flies.

Yes indeed, it's been years!

**Review by themasch — APPROVED** [review]
